dynamo/preprocessing/preprocessor_utils.py

```diff
@@ -1538,3 +1534,116 @@ def pca_selected_genes_wrapper(
     """
 
     adata = pca_monocle(adata, pca_input, n_pca_components=n_pca_components, pca_key=key)
+
+
+def regress_out_parallel(
+    adata: AnnData,
```

Comment:
add a layer argument. by default it is X

Reply:
Added